vermouth/molecule.py

```diff
@@ -511,6 +512,8 @@ def add_or_replace_interaction(self, type_, atoms, parameters, meta=None):
         meta: collections.abc.Mapping
             Metadata for this interaction, such as comments to be written to
             the output.
+        citations: set
+            set of citations that apply when this link is addded to molecule
```

Comment:
s/link/interaction/

But it get worse I now realize: the interaction that's being replaced could have a citation that no longer applies. Leave it for now, but open an issue. I'm (mostly) ok with telling people to cite too many papers.

Reply:
Caching links won't solve this I think. You'd need to track citations per interaction. But I'm ok with erring on the side of caution and giving too many citations.